proxy/http2/Http2ConnectionState.cc

```diff
@@ -1067,16 +1067,33 @@ Http2ConnectionState::find_stream(Http2StreamId id) const
 void
 Http2ConnectionState::restart_streams()
 {
-  Http2Stream *s = stream_list.head;
-
-  while (s) {
-    Http2Stream *next = static_cast<Http2Stream *>(s->link.next);
+  static int x = 0;
+
+  Http2Stream *s   = stream_list.head;
+  Http2Stream *end = s;
+  if (s) {
+    // Change the start point randomly
+    for (int i = x % total_client_streams_count; i; --i) {
```

Comment:
This looks broken. As noted, it is always the case that `x == 0` so the loop does nothing. However, having a loop with a non-zero check to break and *decrementing* the counter is very dangerous - one miss and you've got an infinite loop. Any decrementing loop really needs `i > 0` or `i >= 0`.

Reply:
Fixed.